src/Psalm/Internal/Analyzer/Statements/Expression/Call/Method/ExistingAtomicMethodCallAnalyzer.php

```diff
@@ -415,11 +420,45 @@ public static function analyze(
                 }
             }
 
-            if ($method_storage->assertions) {
+            $assertions = $method_storage->assertions;
```

Comment:
I am not really proud of this but it works, happy to get some ideas on how to properly refactor this to become actually more readable.

Reply:
Another thought here is, why do we even need `inheritdoc`.
I am aware that it might be helpful somewhere (even tho it isn't used anywhere else than in this line), but I'd rather deprecated that property and instead recursively try to detect the inherited parent and use that docblock to merge with the methods one.

This way, every annotation/etc. would be automatically available and usable - no matter if it was inherited or not.